internal/cmd/onepasswordtemplatefuncs.go

```diff
@@ -110,5 +121,79 @@ func getOnepasswordArgs(baseArgs, args []string) []string {
 	if len(args) > 2 {
 		baseArgs = append(baseArgs, "--account", args[2])
 	}
+
 	return baseArgs
 }
+
+// refreshSession will return the current session token if the token within the environment is still valid.
+// Otherwise it will ask the user to sign in and get the new token.
+func (c *Config) onepasswordGetOrRefreshSession(callerArgs []string) string {
+	var account string
+	if len(callerArgs) > 2 {
+		account = callerArgs[2]
+	}
+
+	// maybe there's already a valid token cached in this run for this account
+	token, ok := c.Onepassword.sessionTokens[account]
+	if ok {
+		return token
+	}
+
+	var args []string
+	if account == "" {
+		// If no account has been given then look for any session tokens you can find in the environment
+		token = onepasswordInferSessionToken()
+		args = []string{"signin", "--raw"}
+	} else {
+		token = os.Getenv(fmt.Sprintf(" OP_SESSION_%s", account))
+		args = []string{"signin", account, "--raw"}
+	}
+
+	// Let's not even specify an empty session string if we din't find anything
+	var secretArgs []string
+	if token != "" {
+		secretArgs = []string{"--session", token}
+	}
+
+	name := c.Onepassword.Command
+	// appending the session token here, so we don't log it by accident
+	cmd := exec.Command(name, append(secretArgs, args...)...)
+	cmd.Stdin = c.stdin
+	stderr := &bytes.Buffer{}
+	cmd.Stderr = stderr
+	output, err := cmd.Output()
+	if err != nil {
+		returnTemplateError(fmt.Errorf("%s: %w: %s",
+			shellQuoteCommand(c.Onepassword.Command, args), err, bytes.TrimSpace(stderr.Bytes())))
+		return ""
+	}
+	token = strings.TrimSpace(string(output))
+
+	// cache the session token in memory, so we don't try to refresh it again for this run for this account
+	if c.Onepassword.sessionTokens == nil {
+		c.Onepassword.sessionTokens = make(map[string]string)
+	}
+	c.Onepassword.sessionTokens[account] = token
+
+	return token
+}
+
+// onepasswordInferSessionToken will look for any session tokens in the environment
+// and if it finds exactly one then it will return it.
+func onepasswordInferSessionToken() string {
+	var token string
+	for _, env := range os.Environ() {
+		slice := strings.Split(env, "=")
```

Comment:
nit: Use `chezmoi.CutString` instead of `strings.Split` here. `chezmoi.CutString` is a backport of the `strings.Cut` function coming in Go 1.18.

Reply:
That's neat, I didn't know that was coming in 1.18, thanks. Fixed.